functions/src/controllers/staffs.js

```diff
@@ -0,0 +1,96 @@
+            const { valid, errors } = validateSignUpData(req.body);
+            if (!valid) return validationError(res, errors);
+            const data = await firebase.auth().createUserWithEmailAndPassword(email, password);
+            const token = await data.user.getIdToken(), userId = data.user.uid;
+            const avatar = getFirebaseLink('Headshot-Placeholder-1.png');
+            const userData = createUserData(avatar, email, firstName, lastName, userId, true);
+            await db.doc(`users/${userId}`).set(userData);
+            return successWithData(res, CREATED, `Staff Sign up successful`, token);
+        } catch (error) {
+            return tryCatchError(res, error);
+        }
+    }
+    /**
+	 * Get all staffs, only authorized and authenticated as an admin can do this
+	 * @function
+	 * @param {object} req - request object
+	 * @param {object} res - response object
+	 * @return  {Object} result
+	 */
+    static async allStaffs(req, res) {
+        try {
+            const data = await db.collection('users')
+                .where('isAdmin', '==', true).where('role', '==', 'manager')
+                .orderBy('createdAt', 'asc').get();
+            const docs = data.docs;
+            const staffs = [];
+            for (const doc of docs) {
+                staffs.push(doc.data());
+            }
+            return successNoMessage(res, OK, staffs);
+        } catch (error) {
+            return tryCatchError(res, error);
+        }
+    }
+    /**
+	 * Get staffs by id, only authorized and authenticated as an admin can do this
+	 * @function
+	 * @param {object} req - request object
+	 * @param {object} res - response object
+	 * @return  {Object} result
+	 */
+    static async getOne(req, res) {
+        try {
+            const { id } = req.params;
+            const data = await db.collection('users')
+                .where('userId', '==', id).get();
+            const docs = data.docs;
+            let staff = {};
+            for (const doc of docs) {
+                staff = doc.data();
+            }
+            return successNoMessage(res, OK, staff);
+        } catch (error) {
+            return tryCatchError(res, error);
+        }
+    }
+    /**
+	 * Activate or Deactivate a staff/user,only authorized and authenticated as an admin can do this
+     * @query status = active or inactive
+     * @query type = user or staff
+	 * @function
+	 * @param {object} req - request object
+	 * @param {object} res - response object
+	 * @return  {Object} result
+	 */
+    static async activateOrDeactivateStaff(req, res) {
+        try {
+            const { id } = req.params;
+            const { status, type } = req.query;
+            await db.doc(`/users/${id}`).update({ status: status });
+            return successNoData(res, OK,
+                // eslint-disable-next-line max-len
+                `${type === 'staff' ? 'Staff' : 'User'} ${status === 'active' ? 'activated' : 'deactivated'} successfully`);
```

Comment:
also this is an anti-pattern. Do not nest your ternaries. Use and `if/else if/else` it is clearer.

Reply:
Alright. Will take care of that